lib/exhal/link.ex

```diff
@@ -34,9 +37,37 @@ defmodule ExHal.Link do
   """
   def target_url(a_link, vars \\ %{}) do
     case a_link do
-      %__MODULE__{templated: true} -> {:ok, UriTemplate.expand(a_link.target_url, vars)}
-      %__MODULE__{target_url: nil} -> :error
-      _                      -> {:ok , a_link.target_url}
+      %__MODULE__{href: nil} ->
+        :error
+
+      %__MODULE__{templated: true} ->
+        {:ok, UriTemplate.expand(a_link.href, vars)}
+
+      _ ->
+        {:ok , a_link.href}
+    end
+  end
+
+  @doc """
+  Returns `{:ok, %ExHal.Document{}}`    - representation of the target of the specifyed link
+          `{:error, %ExHal.Document{}}` - non-2XX responses that have a HAL body
+  """
+  def follow(link, vars \\ %{}) do
+    case link do
+      %__MODULE__{target: (t = %Document{})} -> {:ok, t}
```

Comment:
I'm surprised this comment didn't go by the `(ls = [_|[_|_]]) -> ...` pattern in a previous file.